### PR TITLE
fix: Display error message when tier doesn't exist

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes https://github.com/opencollective/opencollective/issues/2362

image

### CODE REVIEW

File: pages/createOrder.js [hunk 1]
```diff
@@ -117,6 +118,20 @@ class CreateOrderPage extends React.Component {
           </MessageBox>
         </Flex>
       );
+    } else if (!data.Tier) {
```
Comment (Betree) [inline pages/createOrder.js]: This is good, but one path is not covered:

If you go to https://opencollective.com/apex/donate there's no tier but it's still valid (it's a custom donation). 

Please test it to ensure that it is correct, but I think you can cover everything by doing:

```suggestion
    } else if (this.props.tierId && !data.Tier) {
```

Comment (JohnTitor) [reply pages/createOrder.js]: I added it to the if condition and a screenshot to PR description.

**Comment by JohnTitor** [comment]
@Betree All the tests are passed, is this okay? If so, I'll squash commits.

File: pages/createOrder.js [hunk 2]
```diff
@@ -117,6 +118,20 @@ class CreateOrderPage extends React.Component {
...
       );
+    } else if (this.props.tierId && !data.Tier) {
+      return (
+        <Flex py={5} justifyContent="center">
+          <MessageBox type="error" withIcon>
+            <FormattedMessage
+              id="createOrder.missingTier"
+              defaultMessage="Oops! This tier doesn't exist or has been removed by the collective admins. "
+            />
+            <Link route="tiers" params={{ collectiveSlug: data.Collective.slug, verb: this.props.verb }}>
```
[the remark this answers is not in the document]
Comment (JohnTitor) [reply pages/createOrder.js]: applied

**Comment by JohnTitor** [comment]
@Betree we can go now :)

**Comment by Betree** [comment]
Thanks for squashing the commits, this is going live :rocket:

**Pull request merged**